### PR TITLE
Data result 

### PR DESCRIPTION
**Pull request by tchasinga**
```
-- Create the vets table
CREATE TABLE vets (
    id SERIAL PRIMARY KEY,
    name VARCHAR(255),
    age INTEGER,
    date_of_graduation DATE
);
```

Screenshot (289)

```
-- Create the species table
CREATE TABLE species (
    id SERIAL PRIMARY KEY,
    name VARCHAR(255)
);
```
Screenshot (290)

```
-- Create the specializations join table
CREATE TABLE specializations (
    vet_id INTEGER REFERENCES vets(id),
    species_id INTEGER REFERENCES species(id),
    PRIMARY KEY (vet_id, species_id)
);
```

Screenshot (291)

```
-- Create the animals table
CREATE TABLE animals (
    id SERIAL PRIMARY KEY,
    name VARCHAR(255)
);
```

Screenshot (292)

```
-- Create the visits join table
CREATE TABLE visits (
    animal_id INTEGER REFERENCES animals(id),
    vet_id INTEGER REFERENCES vets(id),
    visit_date DATE,
    PRIMARY KEY (animal_id, vet_id)
);
```

Screenshot (293)

```
-- Insert data for vets
INSERT INTO vets (name, age, date_of_graduation)
VALUES
    ('William Tatcher', 45, '2000-04-23'),
    ('Maisy Smith', 26, '2019-01-17'),
    ('Stephanie Mendez', 64, '1981-05-04'),
    ('Jack Harkness', 38, '2008-06-08');
```

Screenshot (294)

```
-- Insert data for species
INSERT INTO species (name)
VALUES
    ('Pokemon'),
    ('Digimon');
```
Screenshot (295)

```
-- Insert data for specializations
INSERT INTO specializations (vet_id, species_id)
VALUES
    (1, 1), -- William Tatcher - Pokemon
    (3, 1), -- Stephanie Mendez - Pokemon
    (3, 2), -- Stephanie Mendez - Digimon
    (4, 2); -- Jack Harkness - Digimon
```
Screenshot (296)

```
-- Insert data for animals
INSERT INTO animals (name)
VALUES
    ('Agumon'),
    ('Gabumon'),
    ('Pikachu'),
    ('Devimon'),
    ('Charmander'),
    ('Plantmon'),
    ('Squirtle'),
    ('Angemon'),
    ('Boarmon'),
    ('Blossom');
```

Screenshot (297)

Screenshot (298)

```
INSERT INTO visits (animal_id, vet_id, visit_date)
VALUES
    (1, 1, '2020-05-24'), -- Agumon -> William Tatcher
    (1, 3, '2020-07-22'), -- Agumon -> Stephanie Mendez
    (2, 4, '2021-02-02'), -- Gabumon -> Jack Harkness
    (3, 2, '2020-01-05'), -- Pikachu -> Maisy Smith
    (3, 2, '2020-03-08'), -- Pikachu -> Maisy Smith
    (3, 2, '2020-05-14'), -- Pikachu -> Maisy Smith
    (4, 3, '2021-05-04'), -- Devimon -> Stephanie Mendez
    (5, 4, '2021-02-24'), -- Charmander -> Jack Harkness
    (6, 2, '2019-12-21'), -- Plantmon -> Maisy Smith
    (6, 1, '2020-08-10'), -- Plantmon -> William Tatcher
    (6, 2, '2021-04-07'), -- Plantmon -> Maisy Smith
    (7, 3, '2019-09-29'), -- Squirtle -> Stephanie Mendez
    (8, 4, '2020-10-03'), -- Angemon -> Jack Harkness
    (8, 4, '2020-11-04'), -- Angemon -> Jack Harkness
    (9, 2, '2019-01-24'), -- Boarmon -> Maisy Smith
    (9, 2, '2019-05-15'), -- Boarmon -> Maisy Smith
    (9, 2, '2020-02-27'), -- Boarmon -> Maisy Smith
    (9, 2, '2020-08-03'), -- Boarmon -> Maisy Smith
    (10, 3, '2020-05-24'), -- Blossom -> Stephanie Mendez
    (10, 1, '2021-01-11'); -- Blossom -> William Tatcher

NB : I'VE DECIDED TO USE AN ONLINE IDE TO DISPLAY THE OUTPUT OF THIS COMMAND , BECAUSE MY PGADMIN 4 WAS 
LAUCH AN DIRECT UPDATING 
```

Screenshot (300)
Screenshot (299)

```
-- Who was the last animal seen by William Tatcher?
SELECT a.name AS animal_name
FROM animals a
JOIN visits v ON a.id = v.animal_id
JOIN vets w ON v.vet_id = w.id
WHERE w.name = 'William Tatcher'
ORDER BY v.visit_date DESC
LIMIT 1;

-- How many different animals did Stephanie Mendez see?
SELECT COUNT(DISTINCT v.animal_id) AS num_animals_seen
FROM visits v
JOIN vets s ON v.vet_id = s.id
WHERE s.name = 'Stephanie Mendez';

-- List all vets and their specialties, including vets with no specialties.
SELECT v.name AS vet_name, COALESCE(s.name, 'No Specialization') AS specialization
FROM vets v
LEFT JOIN specializations vs ON v.id = vs.vet_id
LEFT JOIN species s ON vs.species_id = s.id;

-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
SELECT a.name AS animal_name, v.visit_date
FROM animals a
JOIN visits v ON a.id = v.animal_id
JOIN vets s ON v.vet_id = s.id
WHERE s.name = 'Stephanie Mendez'
    AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';

-- What animal has the most visits to vets?
SELECT a.name AS animal_name, COUNT(v.animal_id) AS num_visits
FROM animals a
JOIN visits v ON a.id = v.animal_id
GROUP BY a.name
ORDER BY num_visits DESC
LIMIT 1;

-- Who was Maisy Smith's first visit?
SELECT a.name AS animal_name, v.visit_date
FROM animals a
JOIN visits v ON a.id = v.animal_id
JOIN vets m ON v.vet_id = m.id
WHERE m.name = 'Maisy Smith'
ORDER BY v.visit_date
LIMIT 1;

-- Details for most recent visit: animal information, vet information, and date of visit.
SELECT a.name AS animal_name, v.visit_date, ve.name AS vet_name
FROM animals a
JOIN visits v ON a.id = v.animal_id
JOIN vets ve ON v.vet_id = ve.id
ORDER BY v.visit_date DESC
LIMIT 1;

-- How many visits were with a vet that did not specialize in that animal's species?
SELECT COUNT(*) AS num_visits
FROM visits v
JOIN vets vt ON v.vet_id = vt.id
LEFT JOIN specializations sp ON vt.id = sp.vet_id
WHERE sp.species_id IS NULL OR sp.species_id != v.animal_id;

-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
SELECT s.name AS potential_specialty, COUNT(v.animal_id) AS num_visits
FROM animals a
JOIN visits v ON a.id = v.animal_id
JOIN vets m ON v.vet_id = m.id
LEFT JOIN specializations sp ON m.id = sp.vet_id
LEFT JOIN species s ON sp.species_id = s.id
WHERE m.name = 'Maisy Smith'
GROUP BY s.name
ORDER BY num_visits DESC
LIMIT 1;
```
Screenshot (302)
Screenshot (301)

### CODE REVIEW

**Comment by tchasinga** [comment]
Thanks for this new leaning opportunity  🎉😁
I've learn enough todays

File: schema.sql
Comment (AbelG101) [inline schema.sql:1]: - [ ] Hello @tchasinga, as I've mentioned in the previous milestone all the queries you create should be stored in the `queries.sql` file and not in the PR description. Also make sure only to add screenshots for the queries and not for the create and insert statements.

File: schema.sql [hunk 1]
```diff
@@ -81,3 +81,104 @@ LEFT JOIN animals a ON o.id = a.owner_id
...
+    (6, 2, '2021-04-07'), -- Plantmon -> Maisy Smith
+    (7, 3, '2019-09-29'), -- Squirtle -> Stephanie Mendez
+    (8, 4, '2020-10-03'), -- Angemon -> Jack Harkness
+    (8, 4, '2020-11-04'), -- Angemon -> Jack Harkness
+    (9, 2, '2019-01-24'), -- Boarmon -> Maisy Smith
+    (9, 2, '2019-05-15'), -- Boarmon -> Maisy Smith
+    (9, 2, '2020-02-27'), -- Boarmon -> Maisy Smith
+    (9, 2, '2020-08-03'), -- Boarmon -> Maisy Smith
+    (10, 3, '2020-05-24'), -- Blossom -> Stephanie Mendez
+    (10, 1, '2021-01-11'); -- Blossom -> William Tatcher
```
Comment (AbelG101) [inline schema.sql]: - [ ] Please make sure to put the SQL commands in the proper files as I've noticed the `INSERT` statements are placed inside the `schema.sql` file which is not the proper place
- Everything that creates or changes the table structure should be saved in the schema.sql file.
- Everything that populates tables with data should be saved in the data.sql file.
- Everything that is a query that helps to analyze data should be saved in the queries.sql file.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @tchasinga 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Tables are properly created ✔
- Data is properly inserted ✔
- Correct queries for each question ✔
- Correct use of Git Flow ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql [hunk 2]
```diff
@@ -81,3 +81,109 @@ LEFT JOIN animals a ON o.id = a.owner_id
...
+-- Insert data for visits
+INSERT INTO visits (animal_id, vet_id, visit_date)
+VALUES
+    (1, 1, '2020-05-24'), -- Agumon -> William Tatcher
+    (1, 3, '2020-07-22'), -- Agumon -> Stephanie Mendez
+    (2, 4, '2021-02-02'), -- Gabumon -> Jack Harkness
+    (3, 2, '2020-01-05'), -- Pikachu -> Maisy Smith
+    (3, 2, '2020-03-08'), -- Pikachu -> Maisy Smith
+    (3, 2, '2020-05-14'), -- Pikachu -> Maisy Smith
+    (4, 3, '2021-05-04'), -- Devimon -> Stephanie Mendez
```
Comment (aamirkhan2478) [inline schema.sql]: - [ ] As mentioned by the previous :point_up: reviewer please make sure to put the SQL commands in the proper files as I've noticed the `INSERT` statements are placed inside the `schema.sql` file which is not the proper place but still you didn't place SQL commands in the proper files

- All `CREATE` statements should be inside the `schema.sql` file
- All `INSERT`, and `UPDATE` statements should be inside the `data.sql` file
- All `SELECT` statements should be inside the `queries.sql` file

- [ ] Please check the [requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_join_table.md) and read it carefully

File: queries.sql [hunk 3]
```diff
@@ -114,4 +114,190 @@ GROUP BY species;
...
+    (7, 3, '2019-09-29'), -- Squirtle -> Stephanie Mendez
+    (8, 4, '2020-10-03'), -- Angemon -> Jack Harkness
+    (8, 4, '2020-11-04'), -- Angemon -> Jack Harkness
+    (9, 2, '2019-01-24'), -- Boarmon -> Maisy Smith
+    (9, 2, '2019-05-15'), -- Boarmon -> Maisy Smith
+    (9, 2, '2020-02-27'), -- Boarmon -> Maisy Smith
+    (9, 2, '2020-08-03'), -- Boarmon -> Maisy Smith
+    (10, 3, '2020-05-24'), -- Blossom -> Stephanie Mendez
+    (10, 1, '2021-01-11'); -- Blossom -> William Tatcher
+
```
Comment (aamirkhan2478) [inline queries.sql]: - [ ] As I mentioned above :point_up: please use the proper file for proper SQL statements so kindly create one file called `data.sql` and add all `INSERT` statements inside that file and remove all `INSERT` statements from `queries.sql` and `schema.sql` files :slightly_smiling_face:

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @tchasinga, 



Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Highlights
✔️ Good Effort
✔️ Your design is good.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @tchasinga👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with the join table. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**